### ISSUE TITLE
Calculate and expose a sharpness value in stats

### ISSUE THREAD

**Issue by jumpalottahigh** [lovell/sharp#2251]
Big fan of your work on sharp and thanks for doing what you do and working on it! I've been using it a lot along side Gatsby and Electron projects.

**What are you trying to achieve?**
Have a `sharpness` value in the `stats` object, in order to make assumptions about blurry images.

**Have you searched for similar feature requests?**
I have. Found nothing.

**What would you expect the API to look like?**
```javascript
const stats = await sharp(image)
      .stats()
      .then(stats => stats)
console.log('Sharpness: ', stats.sharpness)
```

What alternatives have you considered?
Calculating myself (or using OpenCV) by converting a single channel to grayscale and calculating a Laplacian. (There are some resources online on how to do this, [here's one for context](https://www.pyimagesearch.com/2015/09/07/blur-detection-with-opencv/))

Is there a sample image that helps explain?
Here is some image from Unsplash:

![sergei-akulich-ZNkvxIPPVeE-unsplash](https://user-images.githubusercontent.com/4155121/84128981-fb14cb80-aa49-11ea-81e1-61ad8a48b8cd.jpg)
Photo by Sergei Akulich on Unsplash

And here is the same image blurred a bit manually in GIMP:
![sergei-akulich-ZNkvxIPPVeE-unsplash-blurred](https://user-images.githubusercontent.com/4155121/84129201-51820a00-aa4a-11ea-96f4-27849fa8ea3d.jpg)

Basically I think it would be a great feature for sharp to have, since it already has low level access to raw image data. I'm not completely sure how hard it would be to implement, so figured I'd open a ticket to see how you feel about it, as well any potential users of the feature.

**Comment by jumpalottahigh** [comment]
@lovell Thanks a lot for this. It works pretty reliably and is about ~ 2x more performant than the OpenCV for NodeJs counterpart.

**Comment by lovell** [comment]
Great, glad it worked, I went ahead and added it to the `stats()` response via commit 8f5495a - this will be in v0.25.4.

**Comment by lovell** [comment]
v0.25.4 now available, thank you for suggesting this improvement.

**Comment by ShinJustinHolly3317** [comment]
Just want to double check.
Does this property work stably so far?
Part of my project needs this feature(very light), but I'm not expertise in image process.
Thank you!

**Comment by lovell** [comment]
@ShinJustinHolly3317 We've seen no problems reported so I've removed the "experimental" flag from the docs via commit https://github.com/lovell/sharp/commit/dd3b78272a76436066c6663a9f6d72a382389fa3

**Comment by Vicrut24** [comment]
Hi Lovell,
Sorry for returning to your comment on on Jun 10, 2020:
Why do they need convolution( with kind of edge enhancement kernel/any) to obtain sharpness in stats?
Looks than it will be there anyway.
Thank you
